### ISSUE TITLE
Restrictions for rerunning partial workflows

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-server#262]
We had started discussions on rerunning workflows here: https://github.com/fractal-analytics-platform/fractal-server/issues/103 and have an overview over submission scenarios here: https://github.com/fractal-analytics-platform/fractal-server/issues/261

While working out the submission scenarios (https://github.com/fractal-analytics-platform/fractal-server/issues/261), the question of restrictions of rerunning partial workflows came up. It should always be possible to rerun the whole workflow from the input files if they are images that were parsed into an OME-Zarr. But if we already start with an OME-Zarr and were rewriting the image data (e.g. illumination correction), then illumination correction cannot be rerun.

Tasks that transform the image data in the OME-Zarr file cannot be rerun arbitrarily. Tasks that create label images or tables can be rerun arbitrarily. Typically, this means that preprocessing & registration cannot be partially rerun, while the image analysis part is more flexible that way (for a workflow phases discussion, see https://github.com/fractal-analytics-platform/fractal-server/issues/14#issuecomment-1313361271 and the illustrations here: https://github.com/fractal-analytics-platform/fractal-server/issues/14#issuecomment-1237802487)

Scenarios:
1) We start with OME-Zarr file => We cannot rerun preprocessing in the same OME-Zarr file (unless we have a backup copy saved somewhere => would need to change input data) [OME-Zarr files as input isn't a common scenario for us yet, but needs to be supported]
2) We started from other files that were converted to OME-Zarr, but they were archived and aren't present anymore => that's a user issue. We should raise a relevant warning & the user restores their files from the archive
3) We started from other files (e.g. PNG, TIFF) and they are still present => we can rerun the whole preprocessing, but not individual tasks. i.e. as long as we rerun the parsing, we can also rerun illumination correction. But we can't run illumination correction a second time on the OME-Zarr file

Here's a sketch for this decision tree:
![IMG_1710](https://user-images.githubusercontent.com/18033446/201625566-810c4e69-6bc8-44df-9b6c-bf82105ab65e.jpeg)

----

tldr: We need to 1) know which tasks transform the images and 2) limit the flexibility by which they can be rerun

**Comment by jluethi** [comment]
Handled in Fractal V2 by tasks having required input types (e.g. illumination correction sets input filter for illum_corr = False)
